social/pipeline/user.py

```diff
@@ -39,17 +50,11 @@ def get_username(strategy, details, user=None, *args, **kwargs):
         else:
             username = uuid4().hex
 
-        short_username = username[:max_length - uuid_length]
         final_username = slug_func(clean_func(username[:max_length]))
 
-        # Generate a unique username for current user using username
-        # as base but adding a unique hash at the end. Original
-        # username is cut to avoid any field max_length.
-        while storage.user.user_exists(username=final_username):
-            username = short_username + uuid4().hex[:uuid_length]
-            final_username = slug_func(clean_func(username[:max_length]))
```

Comment:
What you describe accomplishes the same that's implemented, but doesn't force modification on the User table, which is usually provided by the project, framework, or third-party application (take for instance the commonly used User model in Django projects).

Python-social-auth stores the ``uid``, ``user id`` and ``provider`` is stored in the ``UserSocialAuth`` model (``social_auth_usersocialauth`` DB table), the search is easy on `uid` and `provider`, the related user is just a side effect.